app/routes/api.js

```diff
@@ -12,38 +9,39 @@ const mongoose = require("mongoose"),
   MEME = mongoose.model("MEME"),
   MULTIHIT = mongoose.model("MULTIHIT"),
   Relax = mongoose.model("Relax"),
-  HivTrace = mongoose.model("HivTrace"),
   FADE = mongoose.model("Fade"),
+  API = mongoose.model("API"),
   SLAC = mongoose.model("SLAC");
 
 const shortid = require("shortid"),
   os = require("os"),
   request = require("request"),
+  setup = require("./../../config/setup"),
   logger = require("../../lib/logger");
 
+/*
+ * Submit a job via API
+ */
 function apiSubmit(req, res) {
   logger.info("Incoming request recieved REQ = " + JSON.stringify(req.body));
 
   shortid.characters(
     "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_"
   );
 
-  let postdata = req.body;
-
-  var url_fasta = postdata.fastaLoc,
-    fileName = "api_" + shortid.generate() + "z.nex",
+  var website_url = "datamonkey.org", //Used to build reply URL
+    postdata = req.body,
+    url_fasta = postdata.fastaLoc,
+    today = new Date(),
+    fileName =
+      "api_" +
+      shortid.generate() +
+      today.getMilliseconds() +
+      "." +
+      postdata.fileExtension,
     dest = os.tmpdir(),
     fullFileName = path.join(dest, fileName);
 
-  let datatype = 0,
-    gencodeid = postdata.gencodeid;
-
-  let options = {
-    datatype: 0,
-    gencodeid: gencodeid,
-    mail: postdata.mail,
-  };
-
   function getRequest(url, dest, callback) {
```

Comment:
No filesize limit.

Reply:
Latest update addresses this and adds a file-size limit that is configurable in setup file.